Python/traceback.c

```diff
@@ -554,29 +697,53 @@ tb_displayline(PyTracebackObject* tb, PyObject *f, PyObject *filename, int linen
         if (start_col_byte_offset < 0 || end_col_byte_offset < 0) {
             goto done;
         }
+
         // Convert the utf-8 byte offset to the actual character offset so we
         // print the right number of carets.
-        Py_ssize_t start_offset = _PyPegen_byte_offset_to_character_offset(source_line, start_col_byte_offset);
-        Py_ssize_t end_offset = _PyPegen_byte_offset_to_character_offset(source_line, end_col_byte_offset);
+        Py_ssize_t start_offset = (Py_ssize_t)start_col_byte_offset;
+        Py_ssize_t end_offset = (Py_ssize_t)end_col_byte_offset;
 
-        char offset = truncation;
-        while (++offset <= start_offset) {
-            err = PyFile_WriteString(" ", f);
-            if (err < 0) {
-                goto done;
+        if (source_line) {
+            start_offset = _PyPegen_byte_offset_to_character_offset(source_line, start_col_byte_offset);
+            end_offset = _PyPegen_byte_offset_to_character_offset(source_line, end_col_byte_offset);
+        }
+
+        const char *primary, *secondary;
+        primary = secondary = "^";
+
+        int left_end_offset, right_start_offset;
```

Comment:
Let's spread the assignment over two lines to make this not as long.

Reply:
I first went with it, but then it looked a bit weird since it is not obvious that  `Py_SAFE_DOWNCAST(end_offset, Py_ssize_t, int) - Py_SAFE_DOWNCAST(start_offset, Py_ssize_t, int)` is the initial value. And I didn't want to make a third variable, but yea the line is too long so probably compromise on that.